keras_cv/models/mlp_mixer.py

```diff
@@ -0,0 +1,279 @@
+      classes: optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified.
+      weights: one of `None` (random initialization), or a pretrained
+        weight file path.
+      input_shape: optional shape tuple, defaults to (None, None, 3).
+      pooling: optional pooling mode for feature extraction
+        when `include_top` is `False`.
+        - `None` means that the output of the model will be
+            the 4D tensor output of the
+            last convolutional block.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional block, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+      classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+      name: (Optional) name to pass to the model.  Defaults to "DenseNet".
+
+    Returns:
+      A `keras.Model` instance.
+    """
+    if weights and not tf.io.gfile.exists(weights):
+        raise ValueError(
+            "The `weights` argument should be either "
+            "`None` or the path to the weights file to be loaded. "
+            f"Weights file not found at location: {weights}"
+        )
+
+    if include_top and not classes:
+        raise ValueError(
+            "If `include_top` is True, "
+            "you should specify `classes`. "
+            f"Received: classes={classes}"
+        )
+
+    if input_shape[0] % patch_size[0] != 0:
+        raise ValueError("Input resolution should be divisible" " by the patch size.")
+
+    inputs = layers.Input(shape=input_shape)
+
+    x = inputs
+    if include_rescaling:
+        x = layers.Rescaling(1 / 255.0)(x)
+
+    x = layers.Conv2D(
+        filters=hidden_dim,
+        kernel_size=patch_size,
+        strides=patch_size,
+        padding="valid",
+        name="patchify_and_projection",
+    )(x)
+    x = layers.Reshape((x.shape[1] * x.shape[2], x.shape[3]))(x)
+
+    for i in range(num_blocks):
+        x = MixerBlock(tokens_mlp_dim, channels_mlp_dim, name=f"mixer_block_{i}")(x)
+
+    x = layers.LayerNormalization()(x)
+
+    if include_top:
+        x = layers.GlobalAveragePooling1D(name="avg_pool")(x)
+        x = layers.Dense(classes, activation=classifier_activation, name="predictions")(
+            x
+        )
+
+    elif pooling == "avg":
+        x = layers.GlobalAveragePooling1D(name="avg_pool")(x)
+    elif pooling == "max":
+        x = layers.GlobalMaxPooling1D(name="max_pool")(x)
+
+    model = keras.Model(inputs, x, name=name, **kwargs)
+
+    if weights is not None:
```

Comment:
You mention in this Pr we could port the original weights - do they use the same rescaling scheme we do here?  If so this would be a wonderful correctness check!